modules/serverless-init-containers-apps.adoc

```diff
@@ -0,0 +1,47 @@
+// Module included in the following assemblies:
+//
+// * /serverless/develop/serverless-applications.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-init-containers-apps_{context}"]
+= Configuring init containers
+
+link:https://kubernetes.io/docs/concepts/workloads/pods/init-containers/[Init containers] are specialized containers that are run before application containers in a pod. They are generally used to implement initialization logic for an application, which may include running setup scripts or downloading required configurations.
+
+[NOTE]
+====
+Init containers may cause longer application start-up times and should be used with caution for serverless applications, which are expected to scale up and down frequently.
+====
+
+Multiple init containers are supported in a single Knative service spec. Knative provides a default, configurable naming template if a template name is not provided. The init containers template can be set by adding an appropriate value in a Knative `Service` object spec.
+
+.Prerequisites
+
+* {ServerlessOperatorName} and Knative Serving are installed on your cluster.
+
+* Before you can use init containers for Knative services, an administrator must add the `kubernetes.podspec-init-containers` flag to the `KnativeServing` custom resource (CR). See the {ServerlessProductName} "Global configuration" documentation for more information.
+
+.Procedure
+
+* Add the `initContainers` spec to a Knative `Service` object:
++
+.Example service spec
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+...
+spec:
+  template:
+    spec:
+      initContainers:
+        - imagePullPolicy: Always <1>
```

Comment:
This part is the same as with normal containers however there are some stuff not allowed as described [here](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) (lifecycle, livenessProbe, readinessProbe, or startupProbe ) and Knative validates against them.

Reply:
Since we already link to the Kubernetes docs, I don't think we need to duplicate all of the information that's there, we should just include a minimal example that will work for Knative. Please give an updated example if there's anything else that needs to be shown / explained here in the YAML for this to work.

Reply:
Side note: Serverless applications require a quick scale-up as mentioned in the admin docs. Maybe it would be better to use `imagePullPolicy: IfNotPresent` for the example above. Pulling the image "always" is slowing the scale-up and it's probably not necessary.